core/src/main/scala/org/elasticmq/actor/queue/QueueActorStorage.scala

```diff
@@ -13,8 +13,15 @@ trait QueueActorStorage {
   def copyMessagesToActorRef: Option[ActorRef]
   def moveMessagesToActorRef: Option[ActorRef]
 
+  def nextSequenceNumber(): BigInt = {
+    val next = sequenceNumber
+    sequenceNumber = sequenceNumber + 1
+    next
+  }
+
   var queueData: QueueData = initialQueueData
   var messageQueue: MessageQueue = MessageQueue(queueData.isFifo)
   var fifoMessagesHistory: FifoDeduplicationIdsHistory = FifoDeduplicationIdsHistory.newHistory()
   val receiveRequestAttemptCache = new ReceiveRequestAttemptCache
+  var sequenceNumber = BigInt(0)
```

Comment:
wouldn't a `Long` be enough? :)

Reply:
It probably would, but in specs it says that SequenceNumber should be 128-bit.

Reply:
Hm, if we always use a `Long`, that requirement will be satisfied :)

Also, maybe this var can be private?